.github/workflows/builder.yml

```diff
@@ -91,6 +94,10 @@ jobs:
           twine upload dist/* --skip-existing
 
   build_base:
+    permissions:
+      actions: read # for dawidd6/action-download-artifact to query and download artifacts
+      contents: read # for actions/checkout to fetch code
+      pull-requests: read # for dawidd6/action-download-artifact to query commit hash
```

Comment:
These would not do anything

Reply:
To clarify, do you mean these would not do anything because these are `read` permissions and this is a public repo, so the token will anyways have `read` access? I think that makes sense. 

These changes are automated using https://github.com/step-security/secure-workflows, and that program just takes a workflow as input and fixes it. It does not have context on whether the workflow is in public repo or private repo. For a private repo, these permissions would need to be set explicitly. If you prefer that I remove these from the PR, I can do that. Please let me know.

Reply:
> These changes are automated using [step-security/secure-workflows](https://github.com/step-security/secure-workflows), and that program just takes a workflow as input and fixes it. It does not have context on whether the workflow is in public repo or private repo.

That is why automated tools for things like these are not really helpful and generate tons of noise.

Yes, please address the review comment.

Reply:
Fixed